src/allmydata/web/status.py

```diff
@@ -1172,89 +1225,82 @@ def _get_recent_operations(self):
         recent.reverse()
         return recent
 
-    def render_row(self, ctx, data):
-        s = data
 
-        started_s = render_time(s.get_started())
-        ctx.fillSlots("started", started_s)
+class StatusElement(Element):
+
+    loader = XMLFile(FilePath(__file__).sibling("status.xhtml"))
 
-        si_s = base32.b2a_or_none(s.get_storage_index())
+    def __init__(self, active, recent):
+        super(StatusElement, self).__init__()
+        self._active = active
+        self._recent = recent
+
+    @renderer
+    def active_operations(self, req, tag):
+        active = [self.get_op_state(op) for op in self._active]
+        return SlotsSequenceElement(tag, active)
+
+    @renderer
+    def recent_operations(self, req, tag):
+        active = [self.get_op_state(op) for op in self._recent]
+        return SlotsSequenceElement(tag, active)
+
+    @staticmethod
+    def get_op_state(op):
+        result = dict()
+
+        started_s = render_time(op.get_started())
+        result.update({"started": started_s})
+
```

Comment:
This could just be `result["started"] = started_s` too .. not sure if one has any advantage over the other though..

Reply:
I can't recall why I chose to use `d.update({k:v})` over `d[k] = v`.  Switched to the latter in f9e864c.